service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/diagnosticreport/SpecimenMapper.java

```diff
@@ -166,6 +167,11 @@ private boolean dummySpecimenOrObservationExists(Specimen specimen, List<Observa
         if (specimen.hasCollection()) {
             Specimen.SpecimenCollectionComponent collection = specimen.getCollection();
 
+            if ((specimen.hasReceivedTime() && collection.hasQuantity())
```

Comment:
I think it would be easier to read this way:
```java
if (specimen.hasReceivedTime() && (collection.hasQuantity() || collection.hasBodySite())) {
	specimenNarrativeStatementCommentBuilder.receivedDate(specimen.getReceivedTime());
}
```

Reply:
Refactored, thanks :)